### DISCUSSION TITLE
Using sqlx and pgx in a project

### DISCUSSION THREAD

**Discussion by barbinimichael** [jackc/pgx#1530] [Q&A] [answered]
Hi,

I am looking to use both sqlx and pgx in a project. This would allow users to choose between connection types based on their application needs (requiring postgres-specific functionality or not).
However, one of the points listed in the README was 
```
The pgx interface is recommended when:
1. The application only targets PostgreSQL.
2. No other libraries that require `database/sql` are in use.
```
I was wondering if you could give more clarify on the second point and if there could be potential complications from offering both a sqlx and pgx connection.

Thanks!

**Comment by jackc** [comment, marked as answer]
sqlx requires `database/sql` so you will have to use the `pgx/v5/stdlib` package. There's nothing preventing you from establishing native pgx pools or connections at the same time as `database/sql`.

But it would probably be simpler to use `database/sql` and acquire a pgx connection from it when you need PostgreSQL functionality such as `COPY FROM`. See the package docs for stdlib for an example.
